### ISSUE TITLE
[Bug]: Unable to install version 11.0.11 due to Exception during database setup

### ISSUE THREAD

**Issue by wwidergoldbsc** [pimcore/pimcore#16103]
### Pimcore version

11.0.11

### Steps to reproduce

Running the pimcore-installer

`$ vendor/bin/pimcore-install --no-interaction`

### Actual Behavior

Results in Exception during database setup:

```
Running installation. You can find a detailed install log in var/log/dev.log

Running database setup...

  6/11 [===============>------------]  54%
                                                                                                                          
 [ERROR] The following errors were encountered during installation                                                      
                                                                                                                        

 * An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'x.TABLE_NAME' in 'on clause'
```

With the following Stacktrace:

```
[2023-10-16T11:26:26.798891+02:00] app.ERROR: PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'x.TABLE_NAME' in 'on clause' in /var/www/pimcore/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:130 Stack trace:0 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/PDO/Statement.php(130): PDOStatement->execute(NULL) 
#1 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/Middleware/AbstractStatementMiddleware.php(69): Doctrine\DBAL\Driver\PDO\Statement->execute(NULL) 
#2 /var/www/pimcore/vendor/doctrine/dbal/src/Logging/Statement.php(98): Doctrine\DBAL\Driver\Middleware\AbstractStatementMiddleware->execute(NULL) 
#3 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/Middleware/AbstractStatementMiddleware.php(69): Doctrine\DBAL\Logging\Statement->execute(NULL) 
#4 /var/www/pimcore/vendor/symfony/doctrine-bridge/Middleware/Debug/Statement.php(68): Doctrine\DBAL\Driver\Middleware\AbstractStatementMiddleware->execute(NULL) 
#5 /var/www/pimcore/vendor/doctrine/dbal/src/Connection.php(1098): Symfony\Bridge\Doctrine\Middleware\Debug\Statement->execute() 
#6 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(477): Doctrine\DBAL\Connection->executeQuery('SELECT c.TABLE_...', Array) 
#7 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(561): Doctrine\DBAL\Schema\MySQLSchemaManager->selectTableColumns('pimcore') 
#8 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(408): Doctrine\DBAL\Schema\AbstractSchemaManager->fetchTableColumnsByTable('pimcore') 
#9 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(66): Doctrine\DBAL\Schema\AbstractSchemaManager->doListTables() 
#10 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1651): Doctrine\DBAL\Schema\MySQLSchemaManager->listTables() 
#11 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1663): Doctrine\DBAL\Schema\AbstractSchemaManager->createSchema() 
#12 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(481): Doctrine\DBAL\Schema\AbstractSchemaManager->introspectSchema() 
#13 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(278): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->updateSchema() 
#14 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(667): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->setup() 
#15 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(436): Pimcore\Bundle\InstallBundle\Installer->setupDatabase(Array, Array) 
#16 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(323): Pimcore\Bundle\InstallBundle\Installer->runInstall(Array, Array) 
#17 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Command/InstallCommand.php(390): Pimcore\Bundle\InstallBundle\Installer->install(Array) 
#18 /var/www/pimcore/vendor/symfony/console/Command/Command.php(326): Pimcore\Bundle\InstallBundle\Command\InstallCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#19 /var/www/pimcore/vendor/symfony/console/Application.php(1081): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#20 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(91): Symfony\Component\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#21 /var/www/pimcore/vendor/symfony/console/Application.php(320): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#22 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#23 /var/www/pimcore/vendor/symfony/console/Application.php(174): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#24 /var/www/pimcore/vendor/pimcore/pimcore/bin/pimcore-install(106): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput)) 
#25 /var/www/pimcore/vendor/bin/pimcore-install(120): include('/var/www/pimcor...') 
#26 {main}  Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'x.TABLE_NAME' in 'on clause' in /var/www/pimcore/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28 Stack trace: 
#0 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/PDO/Statement.php(132): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException)) 
#1 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/Middleware/AbstractStatementMiddleware.php(69): Doctrine\DBAL\Driver\PDO\Statement->execute(NULL) 
#2 /var/www/pimcore/vendor/doctrine/dbal/src/Logging/Statement.php(98): Doctrine\DBAL\Driver\Middleware\AbstractStatementMiddleware->execute(NULL) 
#3 /var/www/pimcore/vendor/doctrine/dbal/src/Driver/Middleware/AbstractStatementMiddleware.php(69): Doctrine\DBAL\Logging\Statement->execute(NULL) 
#4 /var/www/pimcore/vendor/symfony/doctrine-bridge/Middleware/Debug/Statement.php(68): Doctrine\DBAL\Driver\Middleware\AbstractStatementMiddleware->execute(NULL) 
#5 /var/www/pimcore/vendor/doctrine/dbal/src/Connection.php(1098): Symfony\Bridge\Doctrine\Middleware\Debug\Statement->execute() 
#6 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(477): Doctrine\DBAL\Connection->executeQuery('SELECT c.TABLE_...', Array) 
#7 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(561): Doctrine\DBAL\Schema\MySQLSchemaManager->selectTableColumns('pimcore') 
#8 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(408): Doctrine\DBAL\Schema\AbstractSchemaManager->fetchTableColumnsByTable('pimcore') 
#9 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(66): Doctrine\DBAL\Schema\AbstractSchemaManager->doListTables() 
#10 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1651): Doctrine\DBAL\Schema\MySQLSchemaManager->listTables() 
#11 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1663): Doctrine\DBAL\Schema\AbstractSchemaManager->createSchema() 
#12 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(481): Doctrine\DBAL\Schema\AbstractSchemaManager->introspectSchema() 
#13 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(278): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->updateSchema() 
#14 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(667): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->setup() 
#15 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(436): Pimcore\Bundle\InstallBundle\Installer->setupDatabase(Array, Array) 
#16 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(323): Pimcore\Bundle\InstallBundle\Installer->runInstall(Array, Array) 
#17 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Command/InstallCommand.php(390): Pimcore\Bundle\InstallBundle\Installer->install(Array) 
#18 /var/www/pimcore/vendor/symfony/console/Command/Command.php(326): Pimcore\Bundle\InstallBundle\Command\InstallCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#19 /var/www/pimcore/vendor/symfony/console/Application.php(1081): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#20 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(91): Symfony\Component\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#21 /var/www/pimcore/vendor/symfony/console/Application.php(320): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#22 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#23 /var/www/pimcore/vendor/symfony/console/Application.php(174): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#24 /var/www/pimcore/vendor/pimcore/pimcore/bin/pimcore-install(106): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput)) 
#25 /var/www/pimcore/vendor/bin/pimcore-install(120): include('/var/www/pimcor...') 
#26 {main}  Next Doctrine\DBAL\Exception\InvalidFieldNameException: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'x.TABLE_NAME' in 'on clause' in /var/www/pimcore/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67 Stack trace: 
#0 /var/www/pimcore/vendor/doctrine/dbal/src/Connection.php(1938): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query)) 
#1 /var/www/pimcore/vendor/doctrine/dbal/src/Connection.php(1880): Doctrine\DBAL\Connection->handleDriverException(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query)) 
#2 /var/www/pimcore/vendor/doctrine/dbal/src/Connection.php(1105): Doctrine\DBAL\Connection->convertExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDO\Exception), 'SELECT c.TABLE_...', Array, Array) 
#3 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(477): Doctrine\DBAL\Connection->executeQuery('SELECT c.TABLE_...', Array) 
#4 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(561): Doctrine\DBAL\Schema\MySQLSchemaManager->selectTableColumns('pimcore') 
#5 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(408): Doctrine\DBAL\Schema\AbstractSchemaManager->fetchTableColumnsByTable('pimcore') 
#6 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/MySQLSchemaManager.php(66): Doctrine\DBAL\Schema\AbstractSchemaManager->doListTables() 
#7 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1651): Doctrine\DBAL\Schema\MySQLSchemaManager->listTables() 
#8 /var/www/pimcore/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1663): Doctrine\DBAL\Schema\AbstractSchemaManager->createSchema() 
#9 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(481): Doctrine\DBAL\Schema\AbstractSchemaManager->introspectSchema() 
#10 /var/www/pimcore/vendor/symfony/doctrine-messenger/Transport/Connection.php(278): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->updateSchema() 
#11 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(667): Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection->setup() 
#12 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(436): Pimcore\Bundle\InstallBundle\Installer->setupDatabase(Array, Array) 
#13 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Installer.php(323): Pimcore\Bundle\InstallBundle\Installer->runInstall(Array, Array) 
#14 /var/www/pimcore/vendor/pimcore/pimcore/bundles/InstallBundle/src/Command/InstallCommand.php(390): Pimcore\Bundle\InstallBundle\Installer->install(Array) 
#15 /var/www/pimcore/vendor/symfony/console/Command/Command.php(326): Pimcore\Bundle\InstallBundle\Command\InstallCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#16 /var/www/pimcore/vendor/symfony/console/Application.php(1081): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#17 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(91): Symfony\Component\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#18 /var/www/pimcore/vendor/symfony/console/Application.php(320): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Pimcore\Bundle\InstallBundle\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#19 /var/www/pimcore/vendor/symfony/framework-bundle/Console/Application.php(80): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#20 /var/www/pimcore/vendor/symfony/console/Application.php(174): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) 
#21 /var/www/pimcore/vendor/pimcore/pimcore/bin/pimcore-install(106): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput)) 
#22 /var/www/pimcore/vendor/bin/pimcore-install(120): include('/var/www/pimcor...') 
#23 {main} [] []
```

Dumping the related SQL Expression shows:
```sql
SELECT c.TABLE_NAME,       c.COLUMN_NAME        AS field,
           IF(
        x.CHECK_CLAUSE IS NOT NULL AND c.COLUMN_TYPE = 'longtext',
        'json',
        c.COLUMN_TYPE
    )       AS type,
       c.IS_NULLABLE        AS `null`,
       c.COLUMN_KEY         AS `key`,
       c.COLUMN_DEFAULT     AS `default`,
       c.EXTRA,
       c.COLUMN_COMMENT     AS comment,
       c.CHARACTER_SET_NAME AS characterset,
       c.COLLATION_NAME     AS collation
FROM information_schema.COLUMNS c
    INNER JOIN information_schema.TABLES t
        ON t.TABLE_NAME = c.TABLE_NAME
    LEFT JOIN information_schema.CHECK_CONSTRAINTS x
    ON (
        c.TABLE_SCHEMA = x.CONSTRAINT_SCHEMA
        AND c.TABLE_NAME = x.TABLE_NAME
        AND x.CHECK_CLAUSE = CONCAT('json_valid(`', c.COLUMN_NAME , '`)')
    ) WHERE c.TABLE_SCHEMA = ? AND t.TABLE_SCHEMA = ? AND t.TABLE_TYPE = 'BASE TABLE' ORDER BY ORDINAL_POSITION
```

### Expected Behavior

Installation exits without error.

**Comment by robertSt7** [comment]
@wwidergoldbsc  Could you please provide proper steps to reproduce it? Just providing `no-interaction` command without required options isn't enough to run installation. Thanks

**Comment by wwidergoldbsc** [comment]
@robertSt7 I have all parameters for the installation prepared as environment variables beforehand, so the pimcore installation should work fine like that.

**Comment by robertSt7** [comment]
@wwidergoldbsc I have tested it with the command from our [docs](https://pimcore.com/docs/platform/Pimcore/Getting_Started/Advanced_Installation_Topics/). Does the error occurs also with that one?

**Comment by turbo-ele** [comment]
@wwidergoldbsc  You might be using mysql but do have mariadb configured? 
```
doctrine:
    dbal:
        connections:
            default:
                server_version: '8.0'
                ...
```

**Comment by wwidergoldbsc** [comment]
Sorry for the late reply, I just got back to it. I can't reproduce it anymore after a fresh setup, I guess there was a problem with my local environment. Sorry for the confusion and trouble. Thanks to everyone for their help!

**Comment by baloraki** [comment]
> Sorry for the late reply, I just got back to it. I can't reproduce it anymore after a fresh setup, I guess there was a problem with my local environment. Sorry for the confusion and trouble. Thanks to everyone for their help!

I have the same problem, can you please explain what you did exactly?

**Comment by wwidergoldbsc** [comment]
@baloraki the problem reappeared recently, but it turned out I had set a wrong serverVersion in my `.env` file.
I was using mysql but had a mariadb serverVersion configured in the database url:

I had to change
`DATABASE_URL=mysql://pimcore:pimcore@database:3306/pimcore?serverVersion=mariadb-10.6.2`
to
`DATABASE_URL=mysql://pimcore:pimcore@database:3306/pimcore?serverVersion=8.0.35`

So @turbo-ele was right, this was just the last place I forgot to check. Thanks again!
